### PR TITLE
Licensing stuff

### PR DESCRIPTION
**Pull request by H-M-H**
This makes the mapinfo-dialog better visible and tries to force mappers to think about licensing. Additionally we now have a tool to extract mapinfo.

### CODE REVIEW

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -5396,6 +5408,10 @@ void CEditorMap::CreateDefault(int EntitiesTexture)
 	MakeGameLayer(new CLayerGame(50, 50));
 	m_pGameGroup->AddLayer(m_pGameLayer);
 
+	str_copy(m_MapInfo.m_aAuthor, g_Config.m_PlayerName, sizeof(m_MapInfo.m_aAuthor));
+	str_copy(m_MapInfo.m_aVersion, "1.0", sizeof(m_MapInfo.m_aVersion));
+	str_copy(m_MapInfo.m_aLicense, "CC BY-SA 4.0", sizeof(m_MapInfo.m_aLicense));
```
Comment (heinrich5991) [inline src/game/editor/editor.cpp]: Does this set a license by default? I'm not sure you can do that.

Comment (H-M-H) [inline src/game/editor/editor.cpp]: it does so, though it is displayed on first mapsave to the mapper.

**Comment by CytraL** [comment]
What license have vanilla teeworlds graphics?

The license is apply to? map data? map graphics (need know what files)? all?

perhaps best use CC0 by default :\

**Comment by H-M-H** [comment]
vanilla: https://www.teeworlds.com/forum/viewtopic.php?pid=116371#p116371
I guess this means public domain.

The main reasons for me creating this were people trying to get their maps away from ddnet and this thread: https://www.teeworlds.com/forum/viewtopic.php?id=11544 (though it only mentions graphics)

**Comment by heinrich5991** [comment]
The vanilla graphics fall under the [zlib license](https://en.wikipedia.org/wiki/Zlib_License) that Teeworlds itself has.

**Comment by H-M-H** [comment]
Well they do not: https://github.com/teeworlds/teeworlds/blob/master/license.txt#L21-L22
They actually are CC-BY-SA 3.0 (at least they were till someone moved data to datasrc xD)

**Comment by heinrich5991** [comment]
Only in 0.7, in 0.6 they're still under the old license: https://github.com/teeworlds/teeworlds/blob/093c1f15db409efbcbb143312f2543c2cd946a1b/license.txt.

**Comment by H-M-H** [comment]
So much confusion about licenses :D.
@heinrich5991 do you still think there should be no default license set even if it is directly displayed to the users ?

**Comment by heinrich5991** [comment]
I don't know, maybe we could make this less default-y. For example, display the map details dialog when the map is about to be saved and show an additional question if the mapper doesn't enter a license. [Continue anyway] [Use CC BY-SA 4.0] [Use CC BY-NC-SA 4.0], maybe with help text when you hover one of the buttons.

(Mh. For some reason I'm suddenly uncertain about this whole thing :o )

**Comment by def-** [comment]
I'm still against this PR, as it makes people realize that they can license their stuff instead of just happily sharing it with everyone, as happens now.

**Comment by Laxa** [comment]
Maybe you can make a ddnet rule that says that ddnet only accepts maps that are under MIT or opensource license ?

**Comment by H-M-H** [comment]
As you see here this is what this commit already proposes: https://github.com/ddnet/ddnet/pull/391/files#diff-623ff8657c88b4e371c870fc321d4e1fR1000

**Comment by def-** [comment]
No one interested in this anymore?

**Pull request closed without merging**